ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2

```diff
@@ -115,27 +115,16 @@ router ospf {{ process_id.id }}
 {%         endif %}
    {{ default_information_originate_cli }}
 {%     endif %}
-{%     if process_id.redistribute.static is defined %}
-{%         set redistribute_static_cli = "redistribute static" %}
-{%         if process_id.redistribute.static.route_map is arista.avd.defined %}
-{%             set redistribute_static_cli = redistribute_static_cli ~ " route-map " ~  process_id.redistribute.static.route_map %}
+{%     for redistribute_route in process_id.redistribute | arista.avd.convert_dicts('source_protocol') | arista.avd.natural_sort('source_protocol') %}
+{%         set redistribute_cli = "redistribute " ~ redistribute_route.source_protocol %}
+{%         if redistribute_route.include_leaked is arista.avd.defined(true) %}
+{%             set redistribute_cli = redistribute_cli ~ " include leaked" %}
```

Comment:
Unfortunately the CLI varies a bit between source protocols. 
https://www.arista.com/en/support/toi/eos-4-24-0f/14464-redistribution-of-leaked-routes-into-igps
Having the dictionary like before, we could implement slight variations. With a list we need to document and guard the supported keys. Ex. we would not support ospf->ospf because we are missing `match <>`, `instance` and the `include leaked` is just `leaked` for ospf->ospf.
I suggest that you document and check for the valid values for "source_protocol" (connected, static, bgp, isis)